packages/vm/src/buildBlock.ts

```diff
@@ -120,7 +120,7 @@ export class BlockBuilder {
       this.vm.common.isActivatedEIP(4844) &&
       typeof this.headerData.excessBlobGas === 'undefined'
     ) {
-      this.headerData.excessBlobGas = opts.parentBlock.header.calcNextExcessBlobGas()
+      this.headerData.excessBlobGas = opts.parentBlock.header.calcNextExcessBlobGas(this.vm.common)
```

Comment:
I think specifically for `buildBlock` we can improve: we can pre-generate the required common for this block (because we have the timestamp) and we should pass this common to calculate the next blob gas.

Reply:
buildblock 's common is correct, its vm's common